test/Microsoft.Extensions.Configuration.Test/ConfigurationTest.cs

```diff
@@ -12,6 +12,86 @@ namespace Microsoft.Extensions.Configuration.Test
     public class ConfigurationTest
     {
         [Fact]
+        public void CanChainConfiguration()
+        {
+            // Arrange
+            var dic1 = new Dictionary<string, string>()
+                {
+                    {"Mem1:KeyInMem1", "ValueInMem1"}
+                };
+            var memConfigSrc1 = new MemoryConfigurationProvider(dic1);
+
+            var configurationBuilder = new ConfigurationBuilder();
+
+            // Act
+            configurationBuilder.Add(memConfigSrc1, load: false);
```

Comment:
@divega probably needs to confirm, I just copied this from another unit test, I would guess if load is false, the data is lazy loaded and only loaded on first call to get as opposed to loaded up front?  It looks like Load is only called by Reload in ConfigurationRoot, but that can't be right for this to be working